integration_tests/src/main/python/limit_test.py

```diff
@@ -28,21 +27,40 @@ def test_simple_limit(data_gen):
             conf = {'spark.sql.execution.sortBeforeRepartition': 'false'})
 
 
-@allow_non_gpu('CollectLimitExec', 'GlobalLimitExec', 'ShuffleExchangeExec')
+# This conf allows CollectLimitExec and GlobalLimitExec to run on GPU
+offset_test_conf = {
+    'spark.rapids.sql.exec.CollectLimitExec': 'true',
+    'spark.rapids.sql.exec.GlobalLimitExec': 'true'
+}
+
+
+@pytest.mark.parametrize('offset', [0, 10, 2048, 4096])
 @pytest.mark.skipif(is_before_spark_340(), reason='offset is introduced from Spark 3.4.0')
-def test_non_zero_offset_fallback():
-    def test_fn(spark):
-        unary_op_df(spark, int_gen, num_slices=1).createOrReplaceTempView("offset_tmp")
-        # `offset` is not exposed to Pyspark, so use the sql string.
-        return spark.sql("select * from offset_tmp limit 10 offset 1")
-
-    assert_gpu_fallback_collect(
-        # We need some processing after the limit to avoid a CollectLimitExec
-        lambda spark: test_fn(spark).repartition(1),
-        'GlobalLimitExec',
-        conf = {'spark.sql.execution.sortBeforeRepartition': 'false'})
-
-    assert_gpu_fallback_collect(
-        test_fn,
-        'CollectLimitExec',
-        conf = {'spark.sql.execution.sortBeforeRepartition': 'false'})
+def test_non_zero_offset(offset):
+    # offset is used in the test cases having no limit
+    # 0: corner case
+    # 1024: offset < df.numRows
+    # 2048: offset = df.numRows
+    # 4096: offset > df.numRows
+
+    offset_sql = "select * from tmp_table offset {}".format(offset)
+    assert_gpu_and_cpu_are_equal_sql(lambda spark: unary_op_df(spark, int_gen, length=2048), 'tmp_table',
+                                     offset_sql, conf=offset_test_conf)
+
+
+@pytest.mark.parametrize('limit, offset', [(0, 0), (0, 10), (1024, 500), (2048, 456), (3000, 111), (500, 500), (100, 600)])
+@pytest.mark.skipif(is_before_spark_340(), reason='offset is introduced from Spark 3.4.0')
+def test_non_zero_offset_with_limit(limit, offset):
+    # In CPU version of spark, (limit, offset) can not be negative number.
+    # Test case description:
+    # (0, 0): Corner case: both limit and offset are 0
+    # (0, 10): Corner case: limit = 0, offset > 0
+    # (1024, 500): offset < limit && limit < df.numRows
+    # (2048, 456): offset < limit && limit = df.numRows
+    # (3000, 111): offset < limit && limit > df.numRows
+    # (500, 500): offset = limit
+    # (100, 600): offset > limit
+
+    limit_offset_sql = "select * from tmp_table limit {} offset {}".format(limit, offset)
+    assert_gpu_and_cpu_are_equal_sql(lambda spark: unary_op_df(spark, int_gen, length=2048), 'tmp_table',
```

Comment:
More than 100 chars at line 47, you can try this

    assert_gpu_and_cpu_are_equal_sql(
        lambda spark: unary_op_df(spark, int_gen, length=2048),
        'tmp_table',
        limit_offset_sql,
        conf=offset_test_conf)